### PR TITLE
Create checkbox component

### PR DESCRIPTION
**Pull request by magwww**
Sign up page checkbox needed validation fix as it was using native validation instead of zod. 
What i did in the proccess:

- Created new checkbox component as it needs different layout than simple input
- Extracted Label to separate component for usability purposes
- Created tailwind classes merging util for classes overwrite
- Added custom validation for checkbox

### CODE REVIEW

File: src/utilities/Label.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      }
+    
+      return (
+        <label {...res} className={cn('inline-block text-md', className)}>
+          {children}
+        </label>
+      );
+};
+
+Label.displayName = 'Label';
```
Comment (javac200) [inline src/utilities/Label.tsx]: Why is this line added?

File: src/utilities/Input.tsx [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
-         rounded-md border-1  px-3.5 py-2 text-black shadow-sm placeholder:text-gray-400 sm:text-sm sm:leading-6 focus:outline-dark-blue"
-        placeholder={props.placeholder}
-        type={props.type}
-        ref={ref}
-        id={props.id}
-        {...props}
-      />
-      {error && <span className="text-alert text-xs">{error.message}</span>}
-  </div>
+export const Input = forwardRef<HTMLInputElement, Props>(({ error, className, ...props }, ref) => (
```
Comment (javac200) [inline src/utilities/Input.tsx]: I wonder why there is a `ref`. I don't see where it is used in the codebase. 
I would remove it. You can do it here or we can create a separate task for this. 
What do you think?

File: src/utilities/Input.tsx [hunk 3]
```diff
@@ -1,26 +1,23 @@
...
+      type={props.type}
+      ref={ref}
+      id={props.id}
+      {...props}
+    />
+    {error && <span className="text-alert text-xs mb-2">{error.message}</span>}
+  </>
 ));
 
 Input.displayName = 'Input';
```
Comment (javac200) [inline src/utilities/Input.tsx]: I would also remove this line ;)

File: src/utilities/Label.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { LabelHTMLAttributes, HTMLAttributes } from "react";
+import { cn } from "../lib/utils/cn";
+
+type Props = LabelHTMLAttributes<HTMLLabelElement> &
+  HTMLAttributes<HTMLParagraphElement>
+
+
+export const Label = ({ htmlFor, children, className, ...res }: Props) => {
+    if (!htmlFor) {
```
Comment (javac200) [inline src/utilities/Label.tsx]: I don't understand why you added this condition. What does it mean? Why does `Label` return `p` tag? It doesn't look good to me.

File: src/utilities/Checkbox.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { FieldError } from "react-hook-form";
+import { Label } from "@utilities/Label";
+import { Input } from "@utilities/Input";
+
+type Props = InputHTMLAttributes<HTMLInputElement> & {
+  label: string;
+  error?: FieldError;
+}
+
+export const Checkbox = forwardRef<HTMLInputElement, Props>(({ error, label, ...props }, ref) => (
```
Comment (javac200) [inline src/utilities/Checkbox.tsx]: The same situation with `ref`.

File: src/utilities/Button.tsx [hunk 6]
```diff
@@ -7,11 +8,11 @@ type Props = ButtonHTMLAttributes<HTMLButtonElement> & {
   textColor: "dark-blue" | "white";
 };
 
-export const Button = ({ text, bg, textColor, type, ...props }: Props) => {
+export const Button = ({ text, bg, textColor, type, className, ...props }: Props) => {
```
Comment (javac200) [inline src/utilities/Button.tsx]: I would replace `className` prop with something different e.g. `classes`. 
`className` is a React key word and it's not recommended to use this for variables/props etc.

File: src/utilities/Input.tsx (hunk 2 above)
Comment (magwww) [reply src/utilities/Input.tsx]: Refs are used here because of react-hook-form. The library needs a reference to the specific input. So we need to keep it that way.

File: src/utilities/Label.tsx (hunk 1 above)
Comment (magwww) [reply src/utilities/Label.tsx]: displayName is added because of the forwardRef. If we use forwardRef (which we need to use - explanation in the other comment), component is called "forwardRef" in react devtools. So it's better to set displayName explicitly for easier debugging.

File: src/utilities/Label.tsx (hunk 4 above)
Comment (magwww) [reply src/utilities/Label.tsx]: Ok, I've changed it to always return label

File: src/utilities/Button.tsx (hunk 6 above)
Comment (magwww) [reply src/utilities/Button.tsx]: So is ref, but we're not using some other word for it, we're just passing a prop called ref, aren't we? I haven't heard about it not being recommended, I'm seeing className used as a prop all the time, even react (legacy) docs are using a variable called className to override it in a component: https://legacy.reactjs.org/docs/faq-styling.html.

File: src/utilities/Label.tsx (hunk 1 above)
Comment (javac200) [reply src/utilities/Label.tsx]: Oh, I've never seen this approach in my commercial apps :D Thanks for explanation

File: src/utilities/Button.tsx (hunk 6 above)
Comment (javac200) [reply src/utilities/Button.tsx]: I found an explanation on airbnb: https://github.com/airbnb/javascript/tree/master/react#naming
It's just a good practice. So it depends on the team, if they want to follow this rule. In my projects, we always used this convention.
I have the same opinion with `ref`. We can create a variable name called `wrapperRef`, `containerRef` etc.

File: src/utilities/Label.tsx (hunk 4 above)
Comment (javac200) [reply src/utilities/Label.tsx]: thanks ;)

File: src/utilities/Input.tsx (hunk 2 above)
Comment (javac200) [reply src/utilities/Input.tsx]: We discussed about it and I mentioned about Controller. In docs, there is an information that it's recommended with external UI component libraries, so I suppose we can skip this: https://react-hook-form.com/get-started#IntegratingwithUIlibraries

File: src/utilities/Button.tsx (hunk 6 above)
Comment (magwww) [reply src/utilities/Button.tsx]: Ok, that convinces me! We might be using their eslint config later, so it's better to stick to their recommendations, I'll change it.

Comment (magwww) [reply src/utilities/Button.tsx]: Although it messes up types a bit, as className is a part of ButtonHTMLAttributes<HTMLButtonElement>, so we need to add classes "by hand".

Comment (aryla12) [reply src/utilities/Button.tsx]: Honestly i will want to talk with you about that because i feel i missing something. For me using the attributes which comes from e.g. ButtonHTMLAttributes is ok 🤔

**Review by aryla12 — APPROVED** [review]

File: src/utilities/Button.tsx (hunk 6 above)
Comment (javac200) [reply src/utilities/Button.tsx]: Actually, I didn't notice before that we use all attributes from ButtonHTMLAttributes as props in  Button component. I've never seen such approach. Could you explain to me why we want to have dozens of props here? This seems like an exaggeration to me.

**Review by javac200 — APPROVED** [review]
